Assets/Scripts/Snake/Snake.cs

```diff
@@ -137,14 +145,33 @@ private void UpdateSpline()
 
             DirectionOnCubeSide stepDirectionOnCubeSide = StepInputDirection.ToLocalDirectionOnCubeSide(ReferenceDirectionForInput);
             CubePoint nextPoint = GetPointOnCubeInDirection(stepDirectionOnCubeSide);
- 
+
             SplinePath.Spline.Add(CalculateSplineKnot(nextPoint));
-            SplinePath.Spline.RemoveAt(0);
-            
             Points.Add(nextPoint);
-            Points.RemoveAt(0);
-
-            UpdateSnakeBody();
+            
+            
+            if (removeNextPoint)
+            {
+                SplinePath.Spline.RemoveAt(0);
+                Points.RemoveAt(0);
+            }
+            else
+            {
+                removeNextPoint = true;
+            }
+            
+            // Check if Snack is on Point
+            if (nextPoint.IsEqual(Snack.Position))
```

Comment:
comment should communicate that snack is getting eaten.
the following actions should also be put into a function called "Eat" or similar